diff --git a/crates/espresso/node-sqlite/src/main.rs b/crates/espresso/node-sqlite/src/main.rs
--- a/crates/espresso/node-sqlite/src/main.rs
+++ b/crates/espresso/node-sqlite/src/main.rs
@@ -1 +1,4 @@
+#[global_allocator]
+static ALLOC: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
+
 pub fn main() -> anyhow::Result<()> {
diff --git a/crates/espresso/node/src/main.rs b/crates/espresso/node/src/main.rs
--- a/crates/espresso/node/src/main.rs
+++ b/crates/espresso/node/src/main.rs
@@ -1 +1,4 @@
+#[global_allocator]
+static ALLOC: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
+
 // NOTE: due to nextest eagerly compiling binaries we allow the build if we're
