diff --git a/crates/espresso/node-sqlite/src/main.rs b/crates/espresso/node-sqlite/src/main.rs
index 1943a65ceef..df53c5a0187 100644
--- a/crates/espresso/node-sqlite/src/main.rs
+++ b/crates/espresso/node-sqlite/src/main.rs
@@ -1 +1,4 @@
+#[global_allocator]
+static ALLOC: snmalloc_rs::SnMalloc = snmalloc_rs::SnMalloc;
+
 pub fn main() -> anyhow::Result<()> {
diff --git a/crates/espresso/node/src/main.rs b/crates/espresso/node/src/main.rs
index 95da67efcd8..836388c0b40 100644
--- a/crates/espresso/node/src/main.rs
+++ b/crates/espresso/node/src/main.rs
@@ -1 +1,4 @@
+#[global_allocator]
+static ALLOC: snmalloc_rs::SnMalloc = snmalloc_rs::SnMalloc;
+
 // NOTE: due to nextest eagerly compiling binaries we allow the build if we're
