diff --git a/crates/espresso/node-sqlite/src/main.rs b/crates/espresso/node-sqlite/src/main.rs
index c7d35e2730..6fc7823a12 100644
--- a/crates/espresso/node-sqlite/src/main.rs
+++ b/crates/espresso/node-sqlite/src/main.rs
@@ -1,3 +1,6 @@
+#[global_allocator]
+static ALLOC: tcmalloc_better::TCMalloc = tcmalloc_better::TCMalloc;
+
 pub fn main() -> anyhow::Result<()> {
     let migrated_envs = espresso_utils::env_compat::migrate_legacy_env_vars();
     tokio::runtime::Runtime::new()?.block_on(espresso_node::main(migrated_envs))
diff --git a/crates/espresso/node/src/main.rs b/crates/espresso/node/src/main.rs
index 5e7c3062c1..c4b826bb12 100644
--- a/crates/espresso/node/src/main.rs
+++ b/crates/espresso/node/src/main.rs
@@ -1,3 +1,6 @@
+#[global_allocator]
+static ALLOC: tcmalloc_better::TCMalloc = tcmalloc_better::TCMalloc;
+
 // NOTE: due to nextest eagerly compiling binaries we allow the build if we're
 // not building with --release (without debug_assertions). There is
 // unfortunately no good way to detect if a build is performed by nextest
diff --git a/crates/espresso/node/src/run.rs b/crates/espresso/node/src/run.rs
index 95801578b6..ae440b0293 100644
--- a/crates/espresso/node/src/run.rs
+++ b/crates/espresso/node/src/run.rs
@@ -18,6 +18,8 @@ use super::{
 use crate::{default_telemetry_endpoint, keyset::KeySet};
 
 pub async fn main(migrated_envs: Vec<(&str, &str)>) -> anyhow::Result<()> {
+    // Spawn tcmalloc's periodic memory-release thread.
+    tcmalloc_better::TCMalloc::process_background_actions_thread();
     let opt = Options::parse();
 
     // Genesis carries the chain ID, which selects the default telemetry
