diff --git a/crates/espresso/node/Cargo.toml b/crates/espresso/node/Cargo.toml
index f253885809..261016c704 100644
--- a/crates/espresso/node/Cargo.toml
+++ b/crates/espresso/node/Cargo.toml
@@ -101,6 +101,7 @@ serde_json = { workspace = true }
 serialization-api = { path = "../../serialization/api" }
 sha2 = { workspace = true }
 snafu = { workspace = true }
+snmalloc-rs = "0.7.4"
 sqlx = { workspace = true }
 staking-cli = { workspace = true }
 static_assertions = { workspace = true }
diff --git a/crates/espresso/node/src/lib.rs b/crates/espresso/node/src/lib.rs
index 58c0745777..7e7ffb34c6 100644
--- a/crates/espresso/node/src/lib.rs
+++ b/crates/espresso/node/src/lib.rs
@@ -1,3 +1,6 @@
+#[global_allocator]
+static ALLOC: snmalloc_rs::SnMalloc = snmalloc_rs::SnMalloc;
+
 mod external_event_handler;
 mod message_compat_tests;
 mod proposal_fetcher;
